### ISSUE TITLE
In-browser?

### ISSUE THREAD

**Issue by srfoster** [wtfaremyinitials/minecraft-schematic#2]
Thanks for the documentation.  That helps a lot.  I've gotten it to work server side.  One thing I wasn't able to do was get it to work on the browser-side.  Example use-case:

1) Do an ajax request to URL for a schematic file,
2) Parse the binary data returned from the request.

Any idea if this should be possible?  I used browserify to bundle up the code.  But I kept getting NBT parse errors when trying to load the schematic in the browser. 

Feel free to close this issue if browser support is beyond the scope of the project.  I do think it would be cool, though.

Oh, and if you're curious, here's the project I'm working on:

http://mod.learntomod.com/schematics/steve-mc1

It's a browser-based level editor.  I'd like to be able to convert traditional NBT schematic files to the format that this level editor is using.  I know I could do that server-side, but doing it browser-side would make things easier.

**Comment by wtfaremyinitials** [comment]
The code definitely should be browserifiable. What does the error look like?

**Comment by srfoster** [comment]
Yeah, I'm probably not using FileReader correctly, or not encoding the string correctly.

Here's a screenshot of the in-browser error message:

https://www.dropbox.com/s/3zebiuzdyherggc/Screenshot%202015-06-15%2010.42.54.png?dl=0

Here's a code snippet.

```

                <input type="file" id="nbtfile" name="file" />
                <input type="submit" onclick="loadNBTFile(document.getElementById('nbtfile
'))" value="Process NBT" />

                <script type="text/javascript">
                        function loadNBTFile(fileSelect) {
                                var file = fileSelect.files[0];

                                var reader = new FileReader();
                                reader.onloadend = function () {
                                  Schematic.loadSchematic(reader.result, function(u, s){
                                     console.log(u)
                                     console.log(s)
                                  })
                                }
                                var nbtfile = reader.readAsBinaryString(file);
                        }
                 </script>

```

**Comment by wtfaremyinitials** [comment]
I'll have a look tomorrow.

**Comment by wtfaremyinitials** [comment]
I think the issue may be that fileReader is reading it as a binary string, and not a nodejs Buffer.

Try this

``` js
Schematic.loadSchematic(new Buffer(reader.result), function(u, s){
    console.log(u)
    console.log(s)
});
```
